MigrationScript/KeyvaultHelperFunctions.ps1

```diff
@@ -37,4 +37,30 @@ function Get-SecretValue($uri)
 	$parsed = $secretValue.value | ConvertFrom-Json
 
     return $parsed
+}
+
+<#
+.SYNOPSIS
+Function to get a delete and purge a secret from keyvault.
+
+.PARAMETER keyVaultName
+Key vault name.
+
+.PARAMETER secretKey
+Secret name to be retrieved from key vault. 
+
+.EXAMPLE
+DeleteAndPurgeSecretFromKeyVault -keyVaultName $keyVaultName -secretKey $name
+#>
+function DeleteAndPurgeSecretFromKeyVault([string]$keyVaultName, [string]$secretKey) {
+	try {
+		Add-KeyVaultRoleAssignment -keyVaultName $keyVaultName;
+		Remove-AzKeyVaultSecret -VaultName $keyVaultName -Name $secretKey -PassThru -Force
+		Start-Sleep -s 15
+		Remove-AzKeyVaultSecret -VaultName $keyVaultName -Name $secretKey -InRemovedState -PassThru -Force;
+		return $true;
+	} catch {
+		$deleteerrMsg = $_.exception.message
+		throw "DeleteAndPurgeSecretFromKeyVault, Failed with error: ($deleteerrMsg)";
```

Comment:
What happens if it fails after add ? If the script is run next time will it fail from Add again ?

Reply:
yeah, working on that still, to retry the add permission and delete secret logic, need to make this a more robust

Reply:
yeah, working on that still, to retry the add permission and delete secret logic, need to make this a more robust